controllers/sr_controller/sr/robot/motor.py

```diff
@@ -37,13 +40,13 @@ def translate(sr_speed_val, sr_motor):
 class Motor:
     """A motor"""
 
-    def __init__(self, m0, m1):
+    def __init__(self, m0: Union[Wheel, LinearMotor], m1: Union[Wheel, Gripper]) -> None:
```

Comment:
i think we can use the base class for this type rather than building a tightly-coupled union

```suggestion
    def __init__(self, m0: MotorBase, m1: MotorBase) -> None:
```

Reply:
I thought this too but unfortunately `MotorBase` doesn't have `set_speed`.
An alternative would be to use a `Protocol` here, but that would add a runtime dependency on `typing-extensions`.

Reply:
eh, leave as-is for now. I'm hoping to try and add some base classes for those motors / sensors properly, at which point we can use _that_ for a type.

Reply:
I noticed this too. Maybe add an abstract method on MotorBase?

Reply:
That's my intention, is to `ABC` the lot of them. But that can come later